pkg/scheduler/queue/user_queues.go

```diff
@@ -19,8 +37,13 @@ type queues struct {
 
 	maxUserQueueSize int
 
-	// Number of connections per querier.
-	querierConnections map[string]int
+	// How long to wait before removing a querier which has got disconnected
+	// but hasn't notified a graceful shutdown.
+	forgetTimeout time.Duration
```

Comment:
Nit: I'd suggest to move `forgetTimeout` to `user_queues.go`, and pass only threshold to `forgetDisconnectQueriers`. `removeQuerierConnection` is also using this field, but only to know whether to keep disconnected querier or not. We could pass that information as a bool to `removeQuerierConnection`. This would simplify `queues` a bit, and allow for easier testing of `forgetDisconnectQueriers`. I would also suggest passing `time.Now()` as parameter to `removeQuerierConnection`, to avoid having dependency on current time in `queues`. WDYT?